src/he.js

```diff
@@ -352,4 +352,4 @@
 		root.he = he;
 	}
 
-}(this));
+}((new Function('return this))());
```

Comment:
Sadly, this won't work with most CSP headers.

Reply:
but at present it don't works with rollup 
(!) `this` has been rewritten to `undefined`
https://rollupjs.org/guide/en#error-this-is-undefined
node_modules/he/he.js
345:   }
346:
347: }(this));

Reply:
@mathiasbynens can you point me to a link where that is documented because at present i know no CSP that can affect that and at present its working well in production code cross browser plattform + even other javascript engines. greetings.

Reply:
@frank-dspeed https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_Security_Policy#eval%28%29_and_friends

Reply:
https://developer.mozilla.org/en-US/docs/Mozilla/Add-ons/WebExtensions/Content_Security_Policy#eval%28%29_and_friends

Reply:
@mathiasbynens so when i understand you right. Your using this only in Web Extension Context that runs under Special CSP? So wehen we want to run it with the changes we need to fork?

Reply:
No, CSP can be used on any web page. Thus, using `Function` is not an option.

Can we look at another way of fixing this?

Reply:
@mathiasbynens the more big problem is not only this lib we use same tech in much bigger JS Frameworks to get a environment agnostic global. Examples Angular, React, we use it all over.  Maybe i need then to write a TC for EMCA/JS to Offer a Shared Global Object with always the same accessor.

But at present in our real world scenarios we have zero issues. In Realworld this method is the best way without 100 of if then checks to get the global object we are running in

Reply:
That proposal already exists: https://github.com/tc39/proposal-global I've implemented it in V8. :)